src/usr/bin/google_set_hostname

```diff
@@ -62,7 +62,8 @@ if [ -n "$new_host_name" ] && [ "${new_host_name%%.*}" != "$old_hostname" ] && !
   # Restart rsyslog to update the hostname.
   systemctl=$(which systemctl 2> /dev/null)
   if [ -x "$systemctl" ]; then
-    hasrsyslog=$($systemctl | grep rsyslog | cut -f1 -d' ')
+	# See man 5 systemd.unit for systemd unit naming rules
+    hasrsyslog=$($systemctl | grep rsyslog | grep -oE '[a-zA-Z0-9:-_.\]+\.(service|socket|device|mount|automount|swap|target|path|timer|slice|scope)')
     if [ ! -z "$hasrsyslog" ]; then
       $systemctl -q --no-block restart "$hasrsyslog"
```

Comment:
If the `try-reload-or-restart` subcommand was used instead of `restart`, would that avoid the need to parse the output of `systemctl`?
```
$systemctl -q --no-block try-reload-or-restart rsyslog.service
```

Reply:
No. The reason we need to parse it isn't the subcommand, it's that the previous code avoided making the assumption that rsyslog was named `rsyslog.service`. If there is some customer using this code on an OS that renamed this to `rsyslogd.service` for some reason this would be a breaking change.

As for why (or whether it was a good idea) to avoid making that assumption, I don't know. But there are a lot of strange edge case issues around this script so I'm hesitant to make new assumptions with it if there's a solution that avoids doing so.